lib/datasource/nuget/v3.ts

```diff
@@ -165,26 +169,43 @@ export async function getReleases(
     releases,
   };
 
-  if (registryUrl.toLowerCase() === defaultNugetFeed.toLowerCase()) {
-    try {
-      const nuspecUrl = `https://api.nuget.org/v3-flatcontainer/${pkgName.toLowerCase()}/${latestStable}/${pkgName.toLowerCase()}.nuspec`;
+  try {
+    const packageBaseAddress = await getResourceUrl(
+      registryUrl,
+      'PackageBaseAddress'
+    );
+    // istanbul ignore else: this is a required v3 api
+    if (is.nonEmptyString(packageBaseAddress)) {
+      const nuspecUrl = `${ensureTrailingSlash(
+        packageBaseAddress
+      )}${pkgName.toLowerCase()}/${latestStable}/${pkgName.toLowerCase()}.nuspec`;
       const metaresult = await http.get(nuspecUrl);
       const nuspec = new XmlDocument(metaresult.body);
       const sourceUrl = nuspec.valueWithPath('metadata.repository@url');
       if (sourceUrl) {
         dep.sourceUrl = sourceUrl;
       }
-    } catch (err) /* istanbul ignore next */ {
+    }
+  } catch (err) /* istanbul ignore next */ {
+    // ignore / silence 404. Seen on proget, if remote connector is used and package is not yet cached
+    if (err instanceof RequestError && err.response?.statusCode === 404) {
       logger.debug(
-        `Cannot obtain sourceUrl for ${pkgName} using version ${latestStable}`
+        { registryUrl, pkgName, pkgVersion: latestStable },
+        `package manifest (.nuspec) not found`
       );
       return dep;
     }
-  } else if (homepage) {
-    dep.sourceUrl = homepage;
+    logger.debug(
+      { err, registryUrl, pkgName, pkgVersion: latestStable },
+      `Cannot obtain sourceUrl`
+    );
+    return dep;
   }
 
+  // istanbul ignore else: not easy testable
   if (homepage) {
+    // only assign if not assigned
+    dep.sourceUrl ??= homepage;
     dep.homepage = homepage;
```

Comment:
There's potential for `homepage` and `sourceUrl` to be different?

```suggestion
    dep.homepage ??= homepage;
```

Reply:
Good catch 👍